marimo/_plugins/ui/_impl/table.py

```diff
@@ -845,15 +849,26 @@ def _get_column_summaries(self, args: EmptyArgs) -> ColumnSummaries:
         # or if we are in stats-only mode,
         # we don't return the chart data
         chart_data = None
+        bin_values: dict[ColumnName, list[BinValue]] = {}
         if (
             self._show_column_summaries != "stats"
             and total_rows <= self._column_charts_row_limit
         ):
-            chart_data, _ = self._to_chart_data_url(self._searched_manager)
+            data = self._searched_manager
+            for column in self._manager.get_column_names():
+                bins = data.get_bin_values(column, DEFAULT_NUM_BINS)
```

Comment:
i wonder if 1) this can be batched to `get_bin_values_for_columns` for better performance and 2) be `lazy` internally. this is so polars (and other dataframes) and do optimizations*

*it may not be needed now (can at least add a comment), since it operates over each column so there may not be any optimizations.